source/parse.h

```diff
@@ -6920,6 +7144,210 @@ class parser
     }
 
 
+    //G requires-expression:
+    //G     'requires' parameter-declaration-list? requirement-body
+    //G
+    //G requirement-body:
+    //G     '{' requirement-seq '}'
+    //G
+    //G requirement-seq:
+    //G     requirement
+    //G     requirement requirement-seq
+    //G
+    //G requirement:
+    //G     nested-requirement
+    //G     negative-requirement
+    //G     simple-requirement
+    //G     type-id ';'          // Cpp1 type-requirement
```

Comment:
I'm thinking that this should simply be
```suggestion
    //G     type-id ';'     // Cpp1 type-requirement
    //G     expression ';'  // Cpp1 simple-requirement
```
Just like the rhs of _is-as-expression_,
potentially template arguments after resolving #727, and
potentially the lhs of _is-as-expression_ after merging #782.

That's consistent, but would also make it easier to reintroduce the pitfalls.
```Cpp2
// https://youtu.be/CXn02MPkn8Y?t=2337
negatable: <T> concept = requires(t: T)
{
  (-t is T); // Hopefully obviously wrong. Should be `{ -t } is T;`.
  -t is T;   // Wrong.
};

// https://youtu.be/CXn02MPkn8Y?t=2418
int_sized: <T> concept = requires(t: T)
{
  (sizeof(T) == 4); // Hopefully obviously wrong. Should be `requires (sizeof(T) == 4);`.
};
// Could also be `int_sized: <T> concept = sizeof(T) == 4;`.

// https://youtu.be/CXn02MPkn8Y?t=2455
nothrow_incrementable: <T> concept = requires(inout t: T)
{
  (noexcept(t++)); // Hopefully obviously wrong. Should be `requires noexcept(t++);` or `{ t++ } !throws;`.
};

// https://quuxplusone.github.io/blog/2021/06/09/another-concepts-chest-mimic/
has_a_but_not_b: <T> concept = requires(t: T)
{
  (a(t));
  !requires (b(t)); // In Cpp2, this works and does the correct thing.
  (!requires (b(t))); // Wrong.
};
```

Reply:
This can still be done for consistency.
And the pitfalls can still be prevent by going back to <https://github.com/hsutter/cppfront/discussions/588#discussioncomment-6726402>.
Which means lowering `-t is T;` to `{ -t is T } -> not_boolean_testable`.
We should also lower `(!requires (b(t)))` similarly,
or all *simple-requirement*s as the linked comment says.